### ISSUE TITLE
Bug in real root isolation

### ISSUE THREAD

**Issue by rprebet** [algebraic-solving/msolve#178]
Hello,

Real root isolation seems to meet a bug on the following input:
```
x,y
0
x^2 - 140*x*y - y^3 + 4900*y^2,
1427247692705959881058285969449495136382746624*y-1
```

Here is the last printed lines
```
Real root isolation starts at precision 142
Degree = 2 	 Max bit size = 451 Min bit size = 163 
nthreads = 1
Bound for positive roots: -141

---!*----!*++*!
Bound for negative roots: -1

!
Time for isolation (elapsed): 0.00 sec


Number of nodes : 12
Number of shifts : 3
Number of half splits : 1
Time in Descartes (elapsed): 0.00 sec
Time in Taylor shifts (elapsed): 0.00 sec

Time for isolation (elapsed): 0.00 sec
Time for refinement (elapsed): 0.00 sec
Number of real roots: 2
Starts real root extraction.
<225><225><225><225><225><225><225><225><225><225><225><225><225><225><225><225><225><225><225><225><225><225><225>Stopped process
```
The process stops by itself after ~10s.

Note that substituting y with its value as following **does not**  fix the problem:
```
x
0
x^2 - 35/356811923176489970264571492362373784095686656*x + 6993513694259203417185601250302526168275458457599/2907354897182427562197295231552018137414565442749272241125960796722557152453591693304764202855054262243050086425064711734138406514458624
```

Many thanks!

**Comment by rprebet** [comment]
Here is another input where the same kind of bug occurs:
```
x,y
0
x^4 + 340*x^3*y + 43350*x^2*y^2 - x^2 + 2456500*x*y^3 - 170*x*y + 52200624*y^4 - 7224*y^2
178405961588244985132285746181186892047843328*y-126152065243154749348922666504749085846203259
'''

**Comment by vneiger** [comment]
> Here is another input where the same kind of bug occurs:

Thanks for the input. Your previous example should be fixed via PR #181 . Could you please check that this also solves this second latest example? (the PR is merged so you can just pull the main branch, recompile, and try your example)

**Comment by ederc** [comment]
I tested the second example provided by @rprebet, it is also fixed via #181. Thus I will close this issue. If you find other bugs, please open a new issue.

**Comment by rprebet** [comment]
This works also for me.
Many thanks!
